WordPress/WordPressTest/PostRepositoryTests.swift

```diff
@@ -0,0 +1,140 @@
+import XCTest
+
+@testable import WordPress
+
+class PostRepositoryTests: CoreDataTestCase {
+
+    private var remoteMock: PostServiceRESTMock!
+    private var repository: PostRepository!
+    private var blogID: CoreDataObjectIdentifier<Blog>!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+
+        let accountService = AccountService(coreDataStack: contextManager)
+        let accountID = accountService.createOrUpdateAccount(withUsername: "username", authToken: "token")
+        try accountService.setDefaultWordPressComAccount(XCTUnwrap(mainContext.existingObject(with: accountID) as? WPAccount))
+
+        let blog = try BlogBuilder(mainContext).withAccount(id: accountID).build()
+
+        contextManager.saveContextAndWait(mainContext)
+
+        blogID = .ofSaved(blog)
+        remoteMock = PostServiceRESTMock()
+        let remoteFactory = PostServiceRemoteFactoryMock()
+        remoteFactory.remoteToReturn = remoteMock
+        repository = PostRepository(coreDataStack: contextManager, remoteFactory: remoteFactory)
+    }
+
+    func testGetPost() async throws {
+        let post = RemotePost(siteID: 1, status: "publish", title: "Post: Test", content: "This is a test post")
+        post?.type = "post"
+        remoteMock.remotePostToReturnOnGetPostWithID = post
+        let postID = try await repository.getPost(withID: 1, from: blogID)
+        let isPage = try await contextManager.performQuery { try postID.existingObject(in: $0) is Page }
+        let title = try await contextManager.performQuery { try postID.existingObject(in: $0).postTitle }
+        let content = try await contextManager.performQuery { try postID.existingObject(in: $0).content }
```

Comment:
Correct, not returning `NSManagedObject` is the only reason. I added [some `performQuery` overloads](https://github.com/wordpress-mobile/WordPress-iOS/pull/21008) not long ago, which makes Swift compiler warns us on some incorrect usages. Those overloads of course won't work for this use case here, because this `performQuery` is a new `async` one. That means we'll need to add the same overloads to the `async performQuery`. But there is also `performAndSave`, so we may need to do the same to `performAndSave` too... Having that many duplicated methods and implementations probably isn't a good idea.

I started to look into writing static analyzer (using tools like SwiftLint, SourceKit, and IndexStore), with the goal that we can parse all `performQuery` and `performAndSave` call sites, find out the closure argument's return type, and report an error if the return type is a `NSManagedObject` (or `Result<NSManagedObject, Error>`, etc.). Having done a little bit research now, I feel like SourceKit may not support that. But I'll continue exploring for a while longer.

Another alternative is, we can potentially use Swift macro to generate those overloads for us. That means, ideally, we'd write and manage the overloads once in the macro, which can be applied to both `performQuery` and `performAndSave`